### ISSUE TITLE
nodes ethx naming

### ISSUE THREAD

**Issue by msimonin** [BeyondTheClouds/enos#16]
Currently ethx names are globally set. If nodes span different cluster vif naming can differ (eth0/eth1 on some nodes and eth1/eth2 on some other for instance). We should probably set this variables on a per host basis instead of globally.

* Inventory file seems a good place to put this information : https://docs.ansible.com/ansible/intro_inventory.html#host-variables 

* Another solution would be to rely on the variable loading mechanisms and place those information on ansible/hosts_vars/<host>/... but will lead to a creation of a bunch of files.

**Comment by asimonet** [comment]
Maybe this should be left to the provider?
On the Grid'5000 side, this information can easily be obtained from the API via Execo.

**Comment by msimonin** [comment]
Yes the provider should get this information and pass it in a way (tbd) to the next phase.

**Comment by msimonin** [comment]
Ok so I would like to make progress on this.
The issue isn't only related on the naming but also on how we are assigning the network interfaces to a role (is it for `api_traffic`, `tunnel_traffic` ...?)

Currently
-  G5K uses max 2 interfaces and assign the second one to external traffic and the first to eveything else
- vagrant boxes have 3 interfaces but only two are used.

A first step could be to let the user decide exactly what he wants in the configuration.globals section  : https://github.com/openstack/kolla-ansible/blob/master/ansible/group_vars/all.yml#L113

To allow this I think we need to invert the two lines in https://github.com/BeyondTheClouds/enos/blob/master/enos/utils/extra.py#L188
to let the config vars override the vars generated by the provider and the framework. The latter will be the default values in the absence of user specified ones.

**Comment by rcherrueau** [comment]
> A first step could be to let the user decide exactly what he wants in the configuration.globals section : https://github.com/openstack/kolla-ansible/blob/master/ansible/group_vars/all.yml#L113

Hum ...

Not sure this works on G5k when you perform a deployment on different clusters?

**Comment by msimonin** [comment]
It doesn't work currently neither ;)

**Comment by msimonin** [comment]
@rcherrueau to address the issue of heterogeneous ethernet devices I propose to postpone it latter. This should be doable by generating dedicated ansible vars in the inventory for each target machines. This logic will be probably implemented by the provider and will provide the defaults mappings to allow a viable deployment. 

Currently we let the framework decide the mapping of the network interfaces to a role (api_traffic, tunnel_traffic) and it's done in a sub-optimal way. IMO we should give the opportunity to the user to make the mapping manually. User defined-mapping will override generated mapping by enos.

**Comment by alebre** [comment]
just a thought....I'm wondering how this has been solved within vm5k (which has been made implemented by Laurent Pouilloux)

**Comment by msimonin** [comment]
For those who have access to the internal document this is sum up here : https://github.com/BeyondTheClouds/Wiki/wiki/13-06-17----refactor-network-interface-mapping

Maybe we should have a specs directory to formalize this kind of change ?
